drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -98,22 +113,29 @@ static int adc_it8xxx2_channel_setup(const struct device *dev,
 	}
 
 	/* The channel is set to ADC alternate function */
-	pinmux_pin_set(config[channel_cfg->channel_id].pinctrls,
-		config[channel_cfg->channel_id].pin,
-		config[channel_cfg->channel_id].alt_fun);
+	pinmux_pin_set(config[channel_id].pinctrls,
+		       config[channel_id].pin,
+		       config[channel_id].alt_fun);
 	LOG_DBG("Channel setup succeeded!");
 	return 0;
 }
 
-static void adc_disable_measurement(void)
+static void adc_disable_measurement(uint32_t ch)
 {
 	struct adc_it8xxx2_regs *const adc_regs = ADC_IT8XXX2_REG_BASE;
 
-	/*
-	 * Disable measurement.
-	 * bit(4:0) = 0x1f : channel disable
-	 */
-	adc_regs->VCH0CTL = IT8XXX2_ADC_DATVAL | IT8XXX2_ADC_CHANNEL_DISABLED;
+	if (ch <= CHIP_ADC_CH7) {
+		/*
+		 * Disable measurement.
+		 * bit(4:0) = 0x1f : channel disable
+		 */
+		adc_regs->VCH0CTL = IT8XXX2_ADC_DATVAL |
+			IT8XXX2_ADC_CHANNEL_DISABLED;
+	} else {
+		/* Channels 13~16 controller setting */
+		adc_regs->adc_vchs_ctrl[ADC_CHANNEL_OFFSET(ch)].VCHCTL =
+			IT8XXX2_ADC_DATVAL;
```

Comment:
Assuming IT8XXX2_ADC_DATVAL is the equivalent  of CHANNEL_DISABLED for 13-16, maybe a comment is in order like line 130 to make this clear

Reply:
Done. Thanks!